R/redcap-event-read.R

```diff
@@ -0,0 +1,193 @@
+#' * `elapsed_seconds`: The duration of the function.
+#' * `raw_text`: If an operation is NOT successful, the text returned by
+#' REDCap.  If an operation is successful, the `raw_text` is returned as an
+#' empty string to save RAM.
+#'
+#' @author
+#' Will Beasley, Ezra Porter
+#'
+#' @references
+#' The official documentation can be found on the 'API Help Page'
+#' and 'API Examples' pages on the REDCap wiki (*i.e.*,
+#' https://community.projectredcap.org/articles/456/api-documentation.html and
+#' https://community.projectredcap.org/articles/462/api-examples.html).
+#' If you do not have an account for the wiki, please ask your campus REDCap
+#' administrator to send you the static material.
+#'
+#' @examples
+#' \dontrun{
+#' uri            <- "https://bbmc.ouhsc.edu/redcap/api/"
+#'
+#' # Query a longitudinal project with a single arm and 3 events
+#' token_1  <- "786334BEB4A87D572DD0E99C4BFCE144"
+#' result_1 <- REDCapR::redcap_arm_export(redcap_uri=uri, token=token_1)
+#' result_1$has_events
+#' result_1$data
+#'
+#' # Query a longitudinal project with 2 arms and complex arm-event mappings
+#' token_2  <- "0434F0E9CF53ED0587847AB6E51DE762"
+#' result_2 <- REDCapR::redcap_arm_export(redcap_uri=uri, token=token_2)
+#' result_2$has_events
+#' result_2$data
+#'
+#' # Query a classic project without events
+#' token_3  <- "D70F9ACD1EDD6F151C6EA78683944E98"
+#' result_3 <- REDCapR::redcap_arm_export(redcap_uri=uri, token=token_3)
+#' result_3$has_events
+#' result_3$data
+#' }
+#' @export
+redcap_event_read <- function(
+    redcap_uri,
+    token,
+    verbose         = TRUE,
+    config_options  = NULL,
+    handle_httr     = NULL
+) {
+
+  checkmate::assert_character(redcap_uri, any.missing=FALSE, len=1, pattern="^.{1,}$")
+  checkmate::assert_character(token     , any.missing=FALSE, len=1, pattern="^.{1,}$")
+
+  token   <- sanitize_token(token)
+  verbose <- verbose_prepare(verbose)
+
+  post_body <-
+    list(
+      token     = token,
+      content   = "event",
+      format    = "csv"
+    )
+
+  try(
+    {
+      # This is the important call that communicates with the REDCap server.
+      kernel <-
+        kernel_api(
+          redcap_uri      = redcap_uri,
+          post_body       = post_body,
+          config_options  = config_options,
+          handle_httr     = handle_httr
+        )
+    },
+    # Don't print the warning in the try block.  Print it below,
+    #   where it's under the control of the caller.
+    silent = TRUE
+  )
+
+  if (exists("kernel")) {
+    if (kernel$success) {
+      has_events <- TRUE
+      outcome_message <- sprintf(
+        paste(
+          "The list of events was retrieved from the REDCap project in %0.1f seconds.",
+          "The http status code was %i."
+        ),
+        kernel$elapsed_seconds,
+        kernel$status_code
+      )
+
+      col_types <- readr::cols(
+        event_name         = readr::col_character(),
+        arm_num            = readr::col_integer(),
```

Comment:
@ezraporter, I'm confused why this is `arm_num` instead of  `arm_id`.  I see the API is returning `arm_num`, but the table's field is `arm_id`.  I guess the PHP code is renaming it.

Note to future self, the API's text is:

```r
Browse[1]> kernel$raw_text
[1] "event_name,arm_num,unique_event_name,custom_event_label,event_id\nIntake,1,intake_arm_1,,12357\nDischage,1,dischage_arm_1,,12358\n\"Follow up\",1,follow_up_arm_1,,12359"
```